### PR TITLE
fix(homework): #MA-588 fix display when a homework has no session

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
When modifying an homework, the selection to modify the date or the session no longer worked.

## Checklist tests
Be able to modify the date of the assignment in session or on a specific date for an upcoming homework.
Not being able to change the date of an homework that was for a past session.
Be able to change the date of an homework to a past date.

## Issue ticket number and link
https://entsupport.gdapublic.fr/browse/MCDT-588

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/model/homework.ts [hunk 1]
```diff
@@ -201,7 +201,9 @@ export class Homework {
 
     init(): void {
         this.type = Mix.castAs(HomeworkType, this.type);
-        this.session = Mix.castAs(Session, this.type);
+        if (this.session) {
+            this.session = Mix.castAs(Session, this.type);
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/homework.ts:205]: Code de la MCDT-503. Rajout du if car si la session est nul, elle ne peut être rattaché a une précédente session.

File: src/main/resources/public/template/homework/homework-form.html [hunk 2]
```diff
@@ -96,7 +96,7 @@ <h2 class="cdt-title-not-clickable">
                         <date-picker class="date-picker-sm margin-bottom-7"  ng-if="homework.attachedToDate"
                                      ng-model="homework.dueDate"  ng-change="syncWorkloadDay()">
                         </date-picker>
-                        <select ng-if="homework.attachedToSession && (isSessionFuture() || !homework.id)" ng-model="homework.session" ng-click="syncWorkloadDay()"
+                        <select ng-if="homework.attachedToSession && ((homework.session && isSessionFuture()) || !homework.id)" ng-model="homework.session" ng-click="syncWorkloadDay()"
```
Comment (BastienBenard) [inline src/main/resources/public/template/homework/homework-form.html:99]: La fonction isSessionFuture retourne vrai si homework.session == null

File: src/main/java/fr/openent/diary/services/impl/HomeworkServiceImpl.java [hunk 3]
```diff
@@ -440,7 +440,7 @@ public void updateHomework(long homeworkId,boolean publishedChanged, JsonObject
         JsonArray values = new JsonArray();
         String query = "UPDATE " + Diary.DIARY_SCHEMA + ".homework " +
                 "SET subject_id = ?, exceptional_label = ?, structure_id = ?, audience_id = ?, estimatedTime = ?," +
-                " color = ?, description = ?, type_id = ?,  modified = NOW() " ;
+                " color = ?, description = ?, type_id = ?, session_id = ?,  modified = NOW() " ;
```
Comment (BastienBenard) [inline src/main/java/fr/openent/diary/services/impl/HomeworkServiceImpl.java:443]: Permet de passer un devoir qui était a la base pour une séance a une date spécifique

File: src/main/java/fr/openent/diary/services/impl/HomeworkServiceImpl.java [hunk 4]
```diff
@@ -454,17 +454,13 @@ public void updateHomework(long homeworkId,boolean publishedChanged, JsonObject
         values.add(homework.getString("color"));
         values.add(homework.getString("description"));
         values.add(homework.getInteger("type_id"));
+        values.add(homework.getInteger("session_id"));
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/diary/services/impl/HomeworkServiceImpl.java:457]: si null => .addNull()

File: src/main/resources/public/ts/model/homework.ts [hunk 5]
```diff
@@ -201,7 +201,9 @@ export class Homework {
 
     init(): void {
         this.type = Mix.castAs(HomeworkType, this.type);
-        this.session = Mix.castAs(Session, this.type);
+        //if the session is not null, it must be detached from its previous session.
+        //if the session is null, it cannot be attached to a previous session.
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/model/homework.ts:205]: Pas compris !

File: src/main/resources/public/ts/model/homework.ts [hunk 6]
```diff
@@ -201,7 +201,9 @@ export class Homework {
...
+        this.session = this.session ? Mix.castAs(Session, this.type) : this.session;
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/homework.ts:206]: Dans la view "Modifier travail à faire" (on parle d'un devoir existante qui est/peut être rattaché à une séance) :
si rattaché -> reset (cf commit MCDT-503 https://github.com/OPEN-ENT-NG/cahier-de-textes/commit/1bf21c1f2bcdd816730d7bca879bf3cffc4915ae#diff-d65747cb90fb0762eb4b807d8fbaddd43888b0df7c5e674521cf27df1f2eda3fR204)

si pas rattaché à une session -> null donc il se passe rien

**Review by JinDouang — APPROVED** [review]

**Pull request merged**